api/v1beta1/foundationdbcluster_types.go

```diff
@@ -2278,3 +2282,20 @@ func (cluster *FoundationDBCluster) GetProcessClassLabels() []string {
 
 	return []string{FDBProcessClassLabel}
 }
+
+// Locality represents the locality for the cluster.
+type Locality struct {
+	Key         string `json:"key,omitempty"`
+	Value       string `json:"value,omitempty"`
+	EnvValue    string `json:"envValue,omitempty"`
+	TopologyKey string `json:"topologyKey,omitempty"`
+}
+
+// GetLocalities returns the cluster localities
+func (cluster *FoundationDBCluster) GetLocalities() []Locality {
+	result := make([]Locality, 0)
+
+	result = append(result, cluster.Spec.Localities...)
+
+	return result
```

Comment:
```suggestion
	localities := make([]Locality, len(cluster.Spec.Localities))
	copy(localities, cluster.Spec.Localities)

	return localities
```

Any reason we want to return a copy here?

Reply:
No strong feelings about it.